Source/WebCore/rendering/RenderListMarker.cpp

```diff
@@ -1843,13 +1843,19 @@ void RenderListMarker::updateContent()
     auto type = style().listStyleType();
     switch (type) {
     // FIXME: handle CSSCounterStyle case rdar://102988393.
-    case ListStyleType::CustomCounterStyle:
     case ListStyleType::String:
         m_textWithSuffix = style().listStyleStringValue();
         m_textWithoutSuffixLength = m_textWithSuffix.length();
         // FIXME: Depending on the string value, we may need the real bidi algorithm.
         m_textIsLeftToRightDirection = u_charDirection(m_textWithSuffix[0]) != U_RIGHT_TO_LEFT;
         break;
+    case ListStyleType::CustomCounterStyle: {
+        auto text = makeString(counterStyle()->prefix(), counterStyle()->text(m_listItem->value()));
+        m_textWithSuffix = makeString(text, counterStyle()->suffix());
+        m_textWithoutSuffixLength = text.length();
+        m_textIsLeftToRightDirection = u_charDirection(text[0]) != U_RIGHT_TO_LEFT;
```

Comment:
is it ok to just checking the first character? (or go with ubidi_getBaseDirection)

Reply:
I've followed the pattern that is already done for all other ListStyleTypes (case default) and case String.

Reply:
sure, though that does not necessarily mean it is correct.

Reply:
oh look, there's a FIXME about this right there.
// FIXME: Depending on the string value, we may need the real bidi algorithm.

Reply:
Perfect! Thanks!